### ISSUE TITLE
cell.fill issue

### ISSUE THREAD

**Issue by ns20011997** [exceljs/exceljs#791]

**Comment by Siemienik** [comment]
@ns20011997 
which version have you used?

**Comment by ns20011997** [comment]
@Siemienik 
latest one -- 1.9.1

**Comment by Siemienik** [comment]
@ns20011997 it is probably a similar issue like here: https://github.com/exceljs/exceljs/pull/722

if I am right, this workaround should works:
```javascript
cell.style = JSON.parse(JSON.stringify(cell.style)); //very bad way to clone anythink, but works.
cell.fill = fill
```

**Comment by vegarringdal** [comment]
hmm, is this because he reads a file and make edits?
Never seen this issue, but maybe because I make the excel file/sheet.

**Comment by DantSu** [comment]
Steps : 
- create a file
- open it with MS excel
- select cells A1 to C5 (It's an example)
- assign a style
- close the file
- open it with exceljs
- run `ws.getCell('B2').font = {...cell.font, color: {argb: 'FF3366FF'}}`
- style of A1 to C5 will be changed

### Workaround

```javascript
const cell = ws.getCell('B2')
cell.style = {
   ...cell.style,
   font: {...cell.font, color: {argb: 'FF3366FF'}}
}
```

### Idea

May be, create a method `setStyle()` on Cell class.

```javascript
setStyle(style) {
  this.style = {...this.style, ...style}
}
```

**Comment by Siemienik** [comment]
@DantSu did you opened PR?

**Comment by DantSu** [comment]
Hi,
If you find that is a good idea too. I will open a pull request.

**Comment by LeanSaucecode** [comment]
> Steps :
> 
> * create a file
> * open it with MS excel
> * select cells A1 to C5 (It's an example)
> * assign a style
> * close the file
> * open it with exceljs
> * run `ws.getCell('B2').font = {...cell.font, color: {argb: 'FF3366FF'}}`
> * style of A1 to C5 will be changed
> 
> ### Workaround
> ```js
> const cell = ws.getCell('B2')
> cell.style = {
>    ...cell.style,
>    font: {...cell.font, color: {argb: 'FF3366FF'}}
> }
> ```
> 
> ### Idea
> May be, create a method `setStyle()` on Cell class.
> 
> ```js
> setStyle(style) {
>   this.style = {...this.style, ...style}
> }
> ```
> 
> #### Use example :
> ```js
> const cell = ws.getCell('B2')
> cell.style = {
>    ...cell.style,
>    font: {...cell.font, color: {argb: 'FF3366FF'}} -- This worked perfectly. Thanks
> }
> ```
> 
> become :
> 
> ```js
> const cell = ws.getCell('B2')
> cell.setStyle({font: {...cell.font, color: {argb: 'FF3366FF'}}})
> ```
> 
> #### Use example :
> ```js
> const cell = ws.getCell('B2')
> cell.style = {
>   ...cell.style,
>   fill: {
>     type: 'pattern',
>     pattern: 'solid',
>     fgColor: {argb: 'FFD8D8D8'}
>   }
> }
> ```
> 
> become :
> 
> ```js
> ws.getCell('B2').setStyle({
>   fill: {
>     type: 'pattern',
>     pattern: 'solid',
>     fgColor: {argb: 'FFD8D8D8'}
>   }
> })
> ```

**Comment by GarryPlays** [comment]
@L-van-der-merwe THANK YOU!!! I went bonkers over the cell style issue. 
Why is your setStyle not a thing yet? I think your proposal sounds great.

**Comment by DantSu** [comment]
> @L-van-der-merwe THANK YOU!!! I went bonkers over the cell style issue. 
> Why is your setStyle not a thing yet? I think your proposal sounds great.

Yes ! I have do this proposal and create a PR but this was never accepted. It's too bad :(

**Comment by Codeonkey** [comment]
Creating the extension method would make the code more readable, it won't increase boilerplate when every extension method returns the cell instance. Unsure why this was never merged, maybe it slipped through their workload.

**Comment by julimuchi** [comment]
THE SOLUTION: https://github.com/exceljs/exceljs/issues/2055
